console-api/proto/consequences.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+
+package rs.tokio.console.consequences;
+
+import "common.proto";
+
+message Causality {
+    oneof update {
+        Extant extant = 1;
+        OpenDirect open_direct = 2;
+        NewIndirect new_indirect = 3;
+        CloseDirect close_direct = 4;
+        CloseIndirect close_indirect = 5;
+        CloseCyclic close_cyclic = 6;
```

Comment:
let's add comments to these explaining what they mean, please? the terminology here is not immediately obvious to me, especially because it's not part of the terminology used in the `tracing` documentation; it would be good to explain what we mean when we say "direct" and "indirect", for example.